applications/nrf5340_audio/src/modules/sd_card.c

```diff
@@ -180,6 +195,136 @@ int sd_card_read(char const *const filename, char *const data, size_t *size)
 	return 0;
 }
 
+int sd_card_segment_open(char const *const filename, char const *const path_to_file)
+{
+	int ret;
+
+	if (seg_read_started) {
+		LOG_ERR("Segment read has already started");
+		/* Which value should I return here? */
+		return -1;
+	}
+
+	if (strlen(path_to_file) + strlen(abs_path_name) > PATH_MAX_LEN) {
+		LOG_ERR("Filepath is too long");
+		return -EINVAL;
+	}
+
+	char abs_path_name[PATH_MAX_LEN + 1] = SD_ROOT_PATH;
+	strcat(abs_path_name, path_to_file);
+	LOG_INF("abs path name:\t%s", abs_path_name);
+
+	if (!sd_init_success) {
```

Comment:
This should likely be the first check in this function. If this is wrong, nothing will work.